crates/bevy_asset/src/io/metadata.rs

```diff
@@ -0,0 +1,58 @@
+/// A enum representing a type of file.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum FileType {
+    Directory,
+    File,
+    // To be compatible with [`std::fs::FileType`].
+    Symlink,
+}
+
+impl FileType {
+    pub fn is_directory(&self) -> bool {
```

Comment:
Since this PR is breaking the API for `AssetIo`, we can consider changing `is_directory` to `is_dir` which is in parallel with `std::fs::FileType`